### ISSUE TITLE
Value cannot be null thrown from cache

### ISSUE THREAD

**Issue by halovanic** [mikeedwards83/Glass.Mapper#283]
I've created a Glass model that has **both** _EnforceTemplate = SitecoreEnforceTemplate.Template_  and _Cachable = true_.

When I make a lookup such as this from the controller and the item does not match the enforced template:

`this.GetContextItem<IMyModel>();`

Then it should return a null object.  However, the cache blows up with this exception:

> [ArgumentNullException: Value cannot be null.
> Parameter name: value]
>    System.Runtime.Caching.MemoryCacheEntry..ctor(String key, Object value, DateTimeOffset absExp, TimeSpan slidingExp, CacheItemPriority priority, Collection`1 dependencies, CacheEntryRemovedCallback removedCallback, MemoryCache cache) +78387
>    System.Runtime.Caching.MemoryCache.AddOrGetExistingInternal(String key, Object value, CacheItemPolicy policy) +563
>    System.Runtime.Caching.ObjectCache.Add(String key, Object value, CacheItemPolicy policy, String regionName) +25
>    Glass.Mapper.Caching.NetMemoryCacheManager.AddOrUpdate(String key, T value) in c:\TeamCity\buildAgent\work\8567e2ba106d3992\Source\Glass.Mapper\Caching\NetMemoryCacheManager.cs:71
>    Glass.Mapper.Pipelines.ObjectConstruction.Tasks.CacheCheck.CacheCheckTask.Execute(ObjectConstructionArgs args) in c:\TeamCity\buildAgent\work\8567e2ba106d3992\Source\Glass.Mapper\Pipelines\ObjectConstruction\Tasks\CacheCheck\CacheCheckTask.cs:50

This seems to be because **NetMemoryCache** (and I believe **HttpCache** too) does not allow null values in its underlying cache implementation.

At a glance, it looks like just a quick null check before the **AddOrUpdate** call in **CacheCheckTask** would be sufficient to get past this issue and get the desired behavior.  I don't think the intent was to cache a null result.  If it was, then we would need some kind of null flag to be saved instead.

Thanks,
Alex

**Comment by mikeedwards83** [comment]
Alex

Thanks for identifying this bug. I have pushed a fix into the development branch and this should be available tomorrow as part of the nightly build.

Mike

**Comment by halovanic** [comment]
Hi Mike,
Thanks for looking into this, your fix looks great to me.  However, I noticed the nightly builds don't seem to be kicking off nightly anymore:

http://build.glass.lu/viewType.html?buildTypeId=bt25&tab=buildTypeStatusDiv shows 7 pending changes.

I also noticed that the pre-release NuGet feed hosted by TeamCity (linked from the Glass download page) doesn't work anymore with Visual Studio 2015/NuGet 3.5.  I think it's likely this bug:
https://github.com/NuGet/Home/issues/2456

Fixing this probably requires the huge hassle of upgrading TeamCity, but perhaps a note on the download page on how to download the artifacts by hand would be sufficient?

Thanks,
Alex
